### PR TITLE
feat: add profile user and logical for this component

### PR DESCRIPTION
**Pull request by ZemaPapenko33**

### CODE REVIEW

File: src/Components/ContentHomePage/ContentHomePage.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import { TContentHomePage } from '../../Types/ComponentTypes'
+import { t } from 'i18next'
+import { Stack } from '@mui/material'
+import Profile from '../Profile/Profile'
+
+const ContentHomePage: React.FC<TContentHomePage> = ({ selectedMenu, userBMI, percentUserBMI }) => {
+  return (
+    <ContentSwitch selectedMenu={selectedMenu}>
+      <Profile data-text={t('t-profile')} userBMI={userBMI} percentUserBMI={percentUserBMI} />
+      <Stack></Stack>
```
Comment (shumak-yaroslav) [inline src/Components/ContentHomePage/ContentHomePage.tsx:12]: what is the empty Stack's?

File: src/Components/Profile/Profile.tsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+          alignItems={'center'}
+          border={0.25}
+          borderRadius={'1rem'}
+          borderColor={theme.palette.blackGrey.main}
+          position="relative"
+        >
+          <CircularProgress
+            variant="determinate"
+            value={100}
+            size={200}
```
Comment (shumak-yaroslav) [inline src/Components/Profile/Profile.tsx:52]: split this component into seperate components

File: src/Pages/HomePage.tsx [hunk 3]
```diff
@@ -14,11 +16,17 @@ const HomePage = (): JSX.Element => {
...
+    percentUserBMI,
     sidebarItemClick,
     logOut
   } = useHomePage()
   useDefaultRedirect()
-  getUser()
+
+  useEffect(() => {
+    getUser()
+    window.history.replaceState({}, '', '/?menu=profile')
```
Comment (shumak-yaroslav) [inline src/Pages/HomePage.tsx:28]: why do we need this replace in `useEffect`, you can add this `searchParam` when you redirect to the `HomePage`, also let's change name of searchParam, `activeMenuItem`, smth like this or you can provide another one

File: src/hooks/use-home-page.hook.ts [hunk 4]
```diff
@@ -30,8 +35,14 @@ function useHomePage(): HomePageHook {
   const email = localStorage.getItem('email')
   const avatarText = userInfo.name[0] + userInfo.surname[0]
   const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null)
-  const [selectedMenu, setSelectedMenu] = useState<string>(`${t('t-profile')}`)
+  const location = useLocation()
+  const queryParams = new URLSearchParams(location.search)
+  const [selectedMenu, setSelectedMenu] = useState<string>(t('t-profile'))
   const open = Boolean(anchorEl)
+  const isTrackingAndTablet = userInfo.notices
+  const userBMI = Math.floor(userInfo.weight / (userInfo.height / 100) ** 2)
```
Comment (shumak-yaroslav) [inline src/hooks/use-home-page.hook.ts:43]: could you avoid of using magic numbers, let's create constants for this numbers

File: src/utils/dataForBMI.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+      break
+    case BMI >= 30.1 && BMI <= 35:
+      BMIData.message = t('t-BMI-30.1-35')
+      BMIData.color = theme.palette.softRed.main
+      break
+    case BMI >= 35.1 && BMI <= 40:
+      BMIData.message = t('t-BMI-35.1-40')
+      BMIData.color = theme.palette.softRed.main
+      break
+    case BMI > 40.1:
```
Comment (shumak-yaroslav) [inline src/utils/dataForBMI.ts:36]: same as above, avoid of using magic numbers

File: src/Pages/HomePage.tsx (hunk 3 above)
Comment (ZemaPapenko33) [reply src/Pages/HomePage.tsx:28]: we need this to set searchParams by default when the application is launched

File: src/hooks/use-home-page.hook.ts (hunk 4 above)
Comment (ZemaPapenko33) [reply src/hooks/use-home-page.hook.ts:43]: i don't understand, that you mean? BMI calculates based on user data

Comment (shumak-yaroslav) [reply src/hooks/use-home-page.hook.ts:43]: in this case magic numbers are 100 and 2

**Review by shumak-yaroslav — APPROVED** [review]

**Pull request merged**